ReactSkia/components/RSkComponentTextInput.cpp

```diff
@@ -263,32 +263,33 @@ void RSkComponentTextInput::processEventKey (rnsKey eventKeyType,bool* stopPropa
   auto textInputEventEmitter = std::static_pointer_cast<TextInputEventEmitter const>(component.eventEmitter);
   auto const &textInputProps = *std::static_pointer_cast<TextInputProps const>(component.props);
   keyPressMetrics.text = RNSKeyMap[eventKeyType];
+  bool bSendOnSelectNotification = false;
+
     //Displayable Charector Range
     if ((eventKeyType >= RNS_KEY_1 && eventKeyType <= RNS_KEY_Less)) {
       if (cursor_.locationFromEnd != 0){
         textString.insert(cursor_.end-cursor_.locationFromEnd,keyPressMetrics.text);
       } else {
         textString = textString+keyPressMetrics.text;
       }
+      cursor_.end = textString.length();
     } else {
       switch(eventKeyType){
         case RNS_KEY_Left:
-          if(cursor_.locationFromEnd < cursor_.end ){
-            RNS_LOG_DEBUG("[processEventKey]Left key pressed cursor_.locationFromEnd = "<<cursor_.locationFromEnd);
-            cursor_.locationFromEnd++; // locationFromEnd
-          }
-          *stopPropagation = true;
-          keyPressMetrics.eventCount = eventCount_;
-          if (!caretHidden_) {
-            drawAndSubmit();
-          }
-          textInputEventEmitter->onKeyPress(keyPressMetrics);
-          *waitForupdateProps = false;
-          return;
         case RNS_KEY_Right:
-          if (cursor_.locationFromEnd>0){
-            RNS_LOG_DEBUG("[processEventKey] Right key pressed cursor_.locationFromEnd = "<<cursor_.locationFromEnd);
-            cursor_.locationFromEnd--;
+          if (RNS_KEY_Left == eventKeyType){
+            if(cursor_.locationFromEnd < cursor_.end ){
+              RNS_LOG_DEBUG("[processEventKey]Left key pressed cursor_.locationFromEnd = "<<cursor_.locationFromEnd);
```

Comment:
1. is it possible to avoid usage of flag here 
2. if there is no change in cursor location , drawAndSubmit also is not required right ? in this case , should we just return ?

Reply:
1. Addressed both the review comments.